room.cpp

```diff
@@ -370,6 +372,16 @@ QList< User* > Room::users() const
     return d->membersMap.values();
 }
 
+QStringList Room::usernames() const {
+	QStringList res;
+
+	for (auto u : d->membersMap.values()) {
+		res.append(u->displayname());
```

Comment:
If you want to use this list for autocompletion, you should use `Room::roomMemberName()` instead of `User::displayname()`. `User::displayname()` is not disambiguated in the current room context so you may get two identically aliased users; `Room::roomMemberName()` takes care of that. Respectively, it's better to name the function `memberNames()`, to better reflect what it returns.